lib/core.ps1

```diff
@@ -665,10 +666,12 @@ function Invoke-ExternalCommand {
                 $s = $_ -replace '(\\+)"', '$1$1"'
                 # escape N consecutive backslash(es), which are at the end of the string, to 2N consecutive ones
                 $s = $s -replace '(\\+)$', '$1$1'
-                # escape double quotes
-                $s = $s -replace '"', '\"'
                 # quote the argument
-                "`"$s`""
+                if ($s -match ' ') {
+                    "`"$s`""
```

Comment:
Do we really need to quote the arg when there is white space in it?

Though the regex introduced above parses args correctly. Quoting the arg can be problematic in some cases and the arguable one could be the `/D` arg of NSIS installers. While using `/D=` to specify install location of a NSIS installer, the path may have white spaces. In this case the arg is quote with double quotes, however, NSIS installer will not use such quoted arg and cause unexpected behaviors - custom install location does not work and software is installed to the default location specified by the installer.

test case:
```sh
# Arg /D is not used by the installer
any-nsis-installer.exe /S "/D=X:\Users\John Wick\scoop\apps\any-nsis\versioned"

# Arg /D is used
any-nsis-installer.exe /S /D=X:\Users\John Wick\scoop\apps\any-nsis\versioned
```

Reply:
I could use regex to just quote the path. Since in Windows, the path must start with `X:` (full path), at least in Scoop, right?

Reply:
According to the [NSIS installer command line usage doc](https://nsis.sourceforge.io/Docs/Chapter3.html), no quote should exist in the arg, therefore quoting the path does not seem to work, I haven't tested it though.

Reply:
Hmm, yes, no quotes should exist here, I've tested by `7tt`

Reply:
But I'm still unsure if spaces without quotes will be recognized correctly by some programs.

Reply:
Maybe an exception can be made for NSIS installer when it's detected? And quotes are still applied in other cases.

Reply:
Done